acme/setup.py

```diff
@@ -22,15 +22,14 @@
 ]
 
 # env markers in extras_require cause problems with older pip: #517
+# Keep in sync with conditional_requirements.py.
 if sys.version_info < (2, 7):
     install_requires.extend([
         # only some distros recognize stdlib argparse as already satisfying
         'argparse',
-        'mock<1.1.0',
```

Comment:
Unfortunately, **we haven't been able to reach a consensus** here yet. In fact, after more extensive IRC discussion, I'm pretty sure **this should not be merged** here.

In general, I strongly believe that:

1. `le-auto` should not impose any constraints on `setup.py`. By the way, how would you impose such constraints on `setup.py` files found external dependencies?

2. Up so far, we have kept our tests together with the main source code. As long as we do it, we should also have test deps in `install_requires`, because that code is shipped to end-users and importable. Rationale for such setup is not really in the scope of this PR, but I'm open to discussion elsewhere. Moreover, messing with this is likely introduce more bugs. That particular concern Erik brought up, was the quickest argument I could come up with, in that particular context.

An obvious solution here is to have `le-auto` pin `"mock==1.1.0"` as this would satisfy `acme`'s setup.py constraints under both Python 2.6 and 2.7. Implementing this plan is about adding one new line to pin mock in `conditional_requirements.py` and one (or more) to add it's hash (anyway it must be easy - it was meant to be this way, because we're likely often to update those pins and their hashes - that's the whole point of this PR). Touching `setup.py` is really the wrong hammer...

Apologies if you think this comes out late in review, @erikrose . This PR was hanging in the queue as WIP almost forever now, lastests commits are still being added, and it was never labelled with `acme` - hence I didn't realize it's touching `acme/` subproject. However, I've hinted you about my concerns much earlier than the comment above.